src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/caseupdated/CaseUpdatedAboutToSubmitHandler.java

```diff
@@ -77,9 +91,59 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
 
         }
 
+        final UserDetails userDetails = idamService.getUserDetails(userAuthorisation);
+        final boolean hasSystemUserRole = userDetails.hasRole(SYSTEM_USER);
+
+        //validate benefit type and dwp issuing office for updateCaseData even triggered by user, which is not by CaseLoader
+        if (!hasSystemUserRole) {
+            validateAndUpdateDwpHandlingOffice(sscsCaseData,preSubmitCallbackResponse);
+        }
+
         return preSubmitCallbackResponse;
     }
 
+    private void validateAndUpdateDwpHandlingOffice(SscsCaseData sscsCaseData, PreSubmitCallbackResponse<SscsCaseData> response) {
+        MrnDetails mrnDetails = sscsCaseData.getAppeal().getMrnDetails();
+        BenefitType benefitType = sscsCaseData.getAppeal().getBenefitType();
+        boolean validBenefitType = validateBenefitType(benefitType,response);
+        boolean validDwpIssuingOffice = validateDwpIssuingOffice(mrnDetails, benefitType, response);
+
+        if (validBenefitType && validDwpIssuingOffice) {
+            String regionalCenter = dwpAddressLookupService.getDwpRegionalCenterByBenefitTypeAndOffice(benefitType.getCode(), mrnDetails.getDwpIssuingOffice());
+            sscsCaseData.setDwpRegionalCentre(regionalCenter);
+        }
+    }
+
+    private boolean validateBenefitType(BenefitType benefitType, PreSubmitCallbackResponse<SscsCaseData> response) {
+        if (benefitType == null || StringUtils.isEmpty(benefitType.getCode())) {
+            response.addWarning("Benefit type code is empty");
+            return false;
+        } else if (Benefit.findBenefitByShortName(benefitType.getCode()).isEmpty()) {
+            String validBenefitTypes = Arrays.stream(Benefit.values()).sequential().map(Benefit::getShortName).collect(Collectors.joining(", "));
```

Comment:
It would be nice to have list of validBenefitTypes string method in Benefit class